### PR TITLE
Feature/config enhanced

### LINKED ISSUE #914: Make apisix's production deployment **FHS**able
Currently, `apisix_home`(`/usr/local/apisix`) is the `prefix` path of `OpenResty`，the configuration, temp data, log are all in this directory.
In `Unix-like` environment, [FHS](https://en.wikipedia.org/wiki/Filesystem_Hierarchy_Standard) is intuitive.
1. Split conf of `apisix` and `openresty`, new directory `/etc/apisix/` for `apisix`'s conf, new directory `/etc/apisix/nginx/` for `OpenResty`.
2. New directory `/var/lib/apisix` as `openresty`'s `prefix`, add `nginx_config.http.temp_directory` config item in `config.yaml`, the default vaule can be `tmp`, and make nginx's `*_temp_path` under this directory.
3. New directory `/var/log/apisix` as `access_log` and `error_log`.
4. As OpenResty's worker process default is ran as `nobody` system user, we don't need create a user.
5. Add `nginx_config.pidfile` config item in `config.yaml`,  the default vaule can `run/apisix.pid`

Production:
```
Configuartion location:
================
/etc/apisix/config.yaml ...
/etc/apisix/nginx/nginx.conf ...

Need config in /etc/apisix/config.yaml:
===========
nginx_config.error_log: /var/log/apisix/error.log
nginx_config.http.access_log: /var/log/apisix/access.log
nginx_config.http.temp_directory: /var/lib/apisix/tmp
nginx_config.pidfile: /var/run/apisix.pid
===========

OpenResty start command line:
=======
openresty -p /var/lib/apisix -c /etc/apisix/nginx/nginx.conf 
```

Development:
Almost keep the same, only add config item for production.

### PR DESCRIPTION
**Pull request by chnliyong**
NOTE: Please read the Contributing.md guidelines before submitting your patch:

https://github.com/apache/incubator-apisix/blob/master/Contributing.md#how-to-add-a-new-feature-or-change-an-existing-one

### Summary
* Separate nginx's configuration to a single directory from `conf`.
* Add config item to support `FHS` install directroy structure.

### Full changelog

1. Separate nginx's configuration to a directory (`conf`  => `conf/nginx`)
2. Add config item `pidfile`, `temp_directory`, which used to generate `nginx.conf` for pid file and the `*_temp_path` directives(`client_body_temp_path`, `fastcgi_temp_path`, `proxy_temp_path`, `scgi_temp_path`, `uwsgi_temp_path`)
3. Modify `nginx.conf` for some production recommendation:
```
log path:
error_log: /var/log/apisix/error.log
access_log: /var/log/apisix/access.log
temp_path: /var/lib/apisix/tmp/
```
4. Modify `config_local.lua` and `id.lua` to support place configuration directory under `/etc/apisix`
5. Modify `bin/apisix` to support `--prefix` and `--config-dir` options for specify nginx's prefix and use `/etc/apisix` as config directory.
### Issues resolved

Fix #914

### CODE REVIEW

File: conf/nginx/nginx.conf [hunk 1]
```diff
@@ -17,9 +17,10 @@
 master_process on;
 
 worker_processes 1;
+worker_cpu_affinity auto;
```
Comment (membphis) [inline conf/nginx/nginx.conf]: We should keep the original content, this file is intended primarily for developers.

File: conf/nginx/nginx.conf [hunk 2]
```diff
@@ -17,9 +17,10 @@
...
 
-error_log logs/error.log warn;
-pid logs/nginx.pid;
+error_log /var/log/apisix/error.log warn;
```
Comment (membphis) [inline conf/nginx/nginx.conf]: ditto

File: conf/nginx/nginx.conf [hunk 3]
```diff
@@ -65,14 +66,24 @@ http {
...
-    access_log logs/access.log main buffer=32768 flush=3;
+    access_log /var/log/apisix/access.log main buffer=32768 flush=3;
     open_file_cache  max=1000 inactive=60;
     client_max_body_size 0;
+    keepalive_timeout 60s;
+    client_header_timeout 60s;
+    client_body_timeout 60s;
+    send_timeout 10s;
+
+    client_body_temp_path /var/lib/apisix/tmp/client_body_temp;
```
Comment (membphis) [inline conf/nginx/nginx.conf]: ditto

File: Makefile [hunk 4]
```diff
@@ -56,7 +56,7 @@ else
 endif
 
 
-### utils:        Installation tools
+### utils:            Installation tools
```
Comment (membphis) [inline Makefile]: Why modify it?

Comment (chnliyong) [reply Makefile]: To make output of `make help` align.

File: Makefile [hunk 5]
```diff
@@ -92,36 +92,41 @@ init: default
...
+	$(OR_EXEC) -p $$PWD/  -c $$PWD/conf/nginx/nginx.conf -s reload
 
 
 ### install:          Install the apisix
 .PHONY: install
 install:
 	$(INSTALL) -d /usr/local/apisix/logs/
-	$(INSTALL) -d /usr/local/apisix/conf/cert
-	$(INSTALL) conf/mime.types /usr/local/apisix/conf/mime.types
+	$(INSTALL) -d /usr/local/apisix/conf/nginx/
```
Comment (moonming) [inline Makefile]: This feature is **not backward compatible** and will affect all users. Is this change really necessary?

Comment (chnliyong) [reply Makefile]: **Backward compatibility**:
1. The `make install` only create `nginx` directory in `/usr/local/apisix`, have no effect for already exist files such as `/usr/local/apisix/conf/cert/*`, `/usr/local/apisix/conf/mine.types`, `/usr/local/apisix/conf/nginx.conf` etc.
2. We use new `bin/apisix` script to `start, stop, restart` apisix, it will automatically generate `nginx.conf` in `/usr/local/apisix/conf/nginx/` directory.

**Necessary**
The root cause I separate this directory is: We don't suggest to modify the files in `nginx` directory, please only change the `*.yaml`.

** Others **
I thought `make install` is for `production` installation, we currently don't supply any upgrade docs for each releases, if we reach `1.0`(may be a GA release), we may need it for each release later. I'm also working on a `recommended` installation scripts, and this PR is to make configuration more flexible.
Thanks!

File: bin/apisix [hunk 6]
```diff
@@ -535,7 +594,9 @@ local function init()
 end
 _M.init = init
 
-local function init_etcd(show_output)
+local function init_etcd()
+    show_output = options["verbose"]
```
Comment (membphis) [inline bin/apisix]: It should be a local variable.
